Formula/gpsd.rb

```diff
@@ -19,4 +19,70 @@ def install
     scons "chrpath=False", "python=False", "strip=False", "prefix=#{prefix}/"
     scons "install"
   end
+
+  def caveats; <<-EOS.undent
+    This Formula for gpsd comes with support for auto-starting gpsd using
+    `brew services start gpsd`. However, this version of gpsd does not
+    automatically detect GPS device addresses.
+
+    After the service is started, you need to use gpsdctl to force gpsd to connect
+    to your GPS. The following tells gpsd to connect to a USB serial device
+    attached to your computer:
+
+      GPSD_SOCKET="#{var}/gpsd.sock" #{HOMEBREW_PREFIX}/sbin/gpsdctl add /dev/tty.usbserial-XYZ
+
+    USB GPS devices use `/dev/tty.usbmodem` or start with `/dev/tty.usbserial-`.
+    Bluetooth GPS devices have a user-defined name starting with `/dev/tty.`.
+
+    For help on using network-connected GPS devices, see the gpsd(8) man page.
+    A list of supported devices and protocols can be seen with:
+
+      #{HOMEBREW_PREFIX}/sbin/gpsd -h
+
+    You can verify that it is working properly with `gpspipe -w` or `cgps`.
+
+    WARNING: gpsd acts independently of macOS' Location Services (CoreLocation).
+    Its privacy controls will not apply to gpsd. Any program that can connect to
+    `localhost` may receive data from your GPS receiver, and thus, your physical
+    location.
+
+    Programs using CoreLocation APIs will not be able to use gpsd by installing
+    this package -- they must explicitly use gpsd APIs.
+    EOS
+  end
+
+  plist_options :manual => "#{HOMEBREW_PREFIX}/sbin/gpsd -N -F #{HOMEBREW_PREFIX}/var/gpsd.sock /dev/tty.usbserial-XYZ"
+
+  def plist; <<-EOS.undent
+    <?xml version="1.0" encoding="UTF-8"?>
+    <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+    <plist version="1.0">
+    <dict>
+      <key>Label</key>
+      <string>#{plist_name}</string>
+      <key>ProgramArguments</key>
+      <array>
+        <string>#{opt_sbin}/gpsd</string>
+        <string>-N</string>
+        <string>-F</string>
+        <string>#{var}/gpsd.sock</string>
+      </array>
+      <key>RunAtLoad</key>
+      <true/>
+      <key>KeepAlive</key>
+      <true/>
+      <key>WorkingDirectory</key>
+      <string>#{HOMEBREW_PREFIX}</string>
+      <key>StandardOutPath</key>
+      <string>#{var}/log/gpsd.log</string>
+      <key>StandardErrorPath</key>
+      <string>#{var}/log/gpsd.log</string>
+    </dict>
+    </plist>
+    EOS
+  end
+
+  test do
+    assert_equal "#{opt_sbin}/gpsd: 3.16 (revision 3.16)", shell_output("#{sbin}/gpsd -V").chomp
```

Comment:
```
Error: gpsd: failed
<"/usr/local/opt/gpsd/sbin/gpsd: 3.16 (revision 3.16)"> expected but was
<"/usr/local/Cellar/gpsd/3.16/sbin/gpsd: 3.16 (revision 3.16)">.
```

Reply:
Seems your bot is silly, and doesn't handle emails properly -- this is still waiting response from you.

Regardless of whether USB autodetection worked on OSX, you would still need
the control socket configured.